hera_mc/cm_hookup.py

```diff
@@ -368,8 +416,31 @@ def is_fully_connected(self, hookup_dict, any_or_all='all'):
         else:
             return num_fully_connected > 0
 
+    def write_hookup_to_file(self, hookup_dict):
+        np.save(self.hookup_local_file, hookup_dict)
+        np.save(self.part_type_cache_file, self.part_type_cache)
+
+    def check_if_hookup_file_is_current(self):
+        import os
+        import time
+        from astropy.time import Time
+        from hera_mc import cm_transfer
+        if os.path.isfile(self.hookup_local_file):
+            result = self.session.query(cm_transfer.CMVersion).order_by(cm_transfer.CMVersion.update_time).all()
+            cm_hash_time = Time(result[-1].update_time, format='gps')
+            stats = os.stat(self.hookup_local_file)
+            file_mod_time = Time(stats.st_mtime, format='unix')
+            return file_mod_time > cm_hash_time
+        else:
+            return False
```

Comment:
Two things could be improved here. First, the `os.stat` call effectively tests whether the file exists anyway, so the `os.path.isfile` call can be avoided. In fact, doing both calls introduces the possibility of a race condition between them, and it is good to develop a habit of spotting and avoiding race conditions since they can lead to subtle but significant bugs.

Second, the logic can be reordered to reduce the amount of rightward drift in the code.

A refined version would look something like:

```python
try:
  stats = os.stat(self.hookup_local_file)
except OSError as e:
  if e.errno == 2:
    return False # ENOENT: file does not exist
  raise # some other error stat'ing the file that we can't handle

result = self.session.query(...)
...
```